src/weights.jl

```diff
@@ -257,35 +257,37 @@ function Base.median{W<:Real}(v::RealVector, w::WeightVec{W})
         isnan(x) && return x
     end
     mask = w.values .!= 0
-    if any(mask)
-        if all(w.values .<= 0)
-            error("no positive weights found")
-        end
-        v = v[mask]
-        wt = w[mask]
-        midpoint = w.sum / 2
-        maxval, maxind = findmax(wt)
-        if maxval > midpoint
-            v[maxind]
-        else
-            permute = sortperm(v)
-            cumulative_weight = zero(eltype(wt))
-            i = 0
-            for (i, p) in enumerate(permute)
-                if cumulative_weight > midpoint
-                    cumulative_weight -= wt[p]
-                    break
-                end
-                cumulative_weight += wt[p]
-            end
+    if !any(mask)
+        error("no nonzero weights found")
```

Comment:
Can we avoid the double negative? I got confused at first by this. Perhaps "at least one weight must be non-zero"?